src/Sulu/Bundle/AdminBundle/Controller/AdminController.php

```diff
@@ -208,7 +209,28 @@ public function contextsAction(Request $request)
         $contexts = $this->adminPool->getSecurityContexts();
         $system = $request->get('system');
 
-        $response = isset($system) ? $contexts[$system] : $contexts;
+        $mappedContexts = [];
+
+        foreach ($contexts as $system => $sections) {
+            foreach ($sections as $section => $contexts) {
```

Comment:
I don't know, I am not a big fan of premature splitting like this... I only edit the lowest level in this array, and I would have to introduce 2 more functions with 3 and 4 parameters, one being a reference being edited, that stuff is even more confusing than the current solution to me... Would something like this be acceptable for you?

```php
foreach ($contexts as $system => $sections) {
            foreach ($sections as $section => $contexts) {
                foreach ($contexts as $context => $permissionTypes) {
                    $mappedContexts[$system][$section][$context] = $this->convertPermissionTypes();
                }
            }
        }
```